src/TopoMojo.Api/Features/User/UserService.cs

```diff
@@ -127,6 +127,17 @@ public async Task<User> AddOrUpdate(UserRegistration model)
             return Mapper.Map<User>(entity);
         }
 
+        public async Task<User> FindByServiceAccountClientId(string clientId)
+        {
+            var user = await _store
+                .DbContext
+                .Users
+                .Where(u => u.ServiceAccountClientId != null && u.ServiceAccountClientId != string.Empty && u.ServiceAccountClientId == clientId)
```

Comment:
Are these null checks necessary? The only time they would matter would be if the passed in clientId were null right? Maybe an explicit check for if clientId is null before even hitting the db would make sense if you need to check for that?